### ISSUE TITLE
In `from` expression, allow `in` to assign to pattern

### ISSUE THREAD

**Issue by julianhyde** [julianhyde/morel#31]
Currently, the left-hand side of `in` must be a variable. We should allow it to be a pattern, to deconstruct rows coming from the right-hand side.

For example, we currently allow
```
from e in emps yield e.deptno + e.sal
```
but we should also allow
```
from {deptno = deptno, sal = sal, ...} in emps yield deptno + sal
```
and we currently allow
```
from p in pairs yield let val (left, right) = p in left + right end
```
but we should also allow
```
from (left, right) in pairs yield left + right
```

**Comment by julianhyde** [comment]
If the pattern contains constant elements then rows that do not match are skipped. For example,
```
= from (left, 2) in [("a", 2), ("b", 4), ("c", 2)]
-   yield left;
val it = ["a","c"] : string list
```

**Comment by julianhyde** [comment]
Fixed in b1f5955.
